File: Swap.cpp
#include<iostream>
using namespace std;

template<typename T> 
    void swapNumber(T &x,T &y){
        T temp = x;
        x = y;
        y = temp;
}

int main(){
    int a = 10,b = 5;
    swapNumber(a,b);
    cout << "Swapped Integer: " << a << " " << b << endl;
    char x = 'a',y='b';
    swapNumber(x,y);
    cout << "Swapped Integer: " << x << " " << y << endl;

}

File: Java/Practise/swap.java
class Swap {
    static void swap(int a,int b){
        // int temp = a;
        // a = b;
        // b = temp;
        a = a^b;
        b = a^b;
        a = a^b;
        System.out.println(a + " " + b);
    }

    public static void main(String[] args) {
        int a = 10;
        int b = 5;
        swap(a,b);
    }
}
